File: src/main/java/plus/maa/backend/controller/request/copilot/CopilotDTO.java
package plus.maa.backend.controller.request.copilot;


import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import plus.maa.backend.repository.entity.Copilot;

import java.util.List;

/**
 * @author LoMu
 * Date  2023-01-10 19:50
 */

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CopilotDTO {

    //关卡名
    @NotBlank(message = "关卡名不能为空")
    private String stageName;

    //难度
    private int difficulty;

    //版本号(文档中说明:最低要求 maa 版本号，必选。保留字段)
    @NotBlank(message = "最低要求 maa 版本不可为空")
    private String minimumRequired;

    //指定干员
    private List<Copilot.Operators> opers;
    //群组
    private List<Copilot.Groups> groups;
    // 战斗中的操作
    private List<Copilot.Action> actions;

    //描述
    private Copilot.Doc doc;
}

File: src/main/kotlin/plus/maa/backend/controller/request/copilot/CopilotDTO.kt
package plus.maa.backend.controller.request.copilot

import jakarta.validation.constraints.NotBlank
import plus.maa.backend.repository.entity.Copilot

/**
 * @author LoMu
 * Date  2023-01-10 19:50
 */
data class CopilotDTO(
    // 关卡名
    @field:NotBlank(message = "关卡名不能为空")
    var stageName: String,
    // 难度
    val difficulty: Int = 0,
    // 版本号(文档中说明:最低要求 maa 版本号，必选。保留字段)
    @field:NotBlank(message = "最低要求 maa 版本不可为空")
    val minimumRequired: String,
    // 指定干员
    val opers: List<Copilot.Operators>? = null,
    // 群组
    val groups: List<Copilot.Groups>? = null,
    // 战斗中的操作
    val actions: List<Copilot.Action>? = null,
    // 描述
    val doc: Copilot.Doc? = null,
    val notification: Boolean = false,
)
